src/map/atcommand.c

```diff
@@ -5277,18 +5277,57 @@ ACMD(follow)
 }
 
 /*==========================================
- * @dropall by [MouseJstr]
- * Drop all your possession on the ground
- *------------------------------------------*/
+* @dropall by [MouseJstr] and [Xantara]
+* Drop all your possession on the ground based on item type
+*------------------------------------------*/
 ACMD(dropall)
 {
+	int type = -1;
+	int count = 0;
+
+	if (message[0]) {
+		type = atoi(message);
+		switch (type) {
```

Comment:
why a switch instead of a range check? this switch can be changed to
```
if (!((type >= IT_HEALING && type <= IT_DELAYCONSUME) || type == IT_CASH || type == -1)) {
	clif->message(fd, msg_fd(fd, 1500));
	clif->message(fd, msg_fd(fd, 1501));
	return false;
}
```

Reply:
copy paste from ragemu, perhaps readability.
anyway, updated to `if (...)`